api/tests/unit/segments/test_unit_segments_services.py

```diff
@@ -0,0 +1,267 @@
+
+    # When
+    delete_segment(segment, author=author)
+
+    # Then
+    for rule_id in rule_ids:
+        rule = SegmentRule.objects.get(pk=rule_id)
+        assert rule.deleted_at is not None
+
+
+def test_delete_segment_soft_deletes_all_conditions(
+    project: Project, admin_user: FFAdminUser
+) -> None:
+    # Given
+    segment = _create_segment_with_nested_rules(project)
+    condition_ids = list(
+        Condition.objects.filter(rule__rule__rule__segment=segment).values_list(
+            "id", flat=True
+        )
+    )
+    author = AuthorData(user=admin_user)
+
+    # When
+    delete_segment(segment, author=author)
+
+    # Then
+    for condition_id in condition_ids:
+        condition = Condition.objects.get(pk=condition_id)
+        assert condition.deleted_at is not None
+
+
+def test_delete_segment_creates_audit_log(
+    project: Project, admin_user: FFAdminUser
+) -> None:
+    # Given
+    segment = Segment.objects.create(name="Test Segment", project=project)
+    segment_id = segment.id
+    segment_uuid = str(segment.uuid)
+    author = AuthorData(user=admin_user)
+
+    # When
+    delete_segment(segment, author=author)
+
+    # Then
+    audit_log = AuditLog.objects.filter(
+        related_object_id=segment_id,
+        related_object_type=RelatedObjectType.SEGMENT.name,
+    ).first()
+
+    assert audit_log is not None
+    assert audit_log.log == SEGMENT_DELETED_MESSAGE % "Test Segment"
+    assert audit_log.author == admin_user
+    assert audit_log.project == project
+    assert audit_log.related_object_uuid == segment_uuid
+
+
+def test_delete_segment_deletes_all_versions(
+    project: Project, admin_user: FFAdminUser
+) -> None:
+    # Given
+    segment = Segment.objects.create(name="Test Segment", project=project)
+    revision = segment.clone(is_revision=True)
+    author = AuthorData(user=admin_user)
+
+    # When
+    delete_segment(segment, author=author)
+
+    # Then
+    segment.refresh_from_db()
+    revision.refresh_from_db()
+    assert segment.deleted_at is not None
+    assert revision.deleted_at is not None
+
+
+def test_delete_segment_query_count_is_constant(
+    project: Project, admin_user: FFAdminUser
+) -> None:
+    # Given
+    small_segment = _create_segment_with_nested_rules(
+        project, num_rules=2, num_nested=2, num_conditions=2
+    )
+    large_segment = _create_segment_with_nested_rules(
+        project, num_rules=10, num_nested=5, num_conditions=5
+    )
+    author = AuthorData(user=admin_user)
+
+    # When
+    reset_queries()
+    with CaptureQueriesContext(connection) as ctx_small:
+        delete_segment(small_segment, author=author)
+    small_query_count = len(ctx_small.captured_queries)
+
+    reset_queries()
+    with CaptureQueriesContext(connection) as ctx_large:
+        delete_segment(large_segment, author=author)
+    large_query_count = len(ctx_large.captured_queries)
+
+    # Then
+    assert small_query_count == large_query_count
```

Comment:
Don't we know what the number is here too? Can't we do something like `small_query_count == large_query_count == x`?